From 24d2a42571f3928c8d70ba504abc741e5b8a4f0a Mon Sep 17 00:00:00 2001
From: Liam Girdwood <lrg@ti.com>
Date: Thu, 8 Jan 2015 14:12:33 +0530
Subject: [PATCH 05/24] ASoC: DAPM: Use snd_soc_get_enum_text()

Issue: GMINL-3680

Use snd_soc_get_enum_text() instead of soc_enum texts.
As texts can be NULL, in certain scenarios

Change-Id: I09941b0c68656fb236af810837913a0e0c50b746
Signed-off-by: Liam Girdwood <lrg@ti.com>
Signed-off-by: Samreen Nilofer <samreen.nilofer@intel.com>
---
 sound/soc/soc-dapm.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/sound/soc/soc-dapm.c b/sound/soc/soc-dapm.c
index 93ce561..cf8788d 100644
--- a/sound/soc/soc-dapm.c
+++ b/sound/soc/soc-dapm.c
@@ -567,7 +567,8 @@ static void dapm_set_path_status(struct snd_soc_dapm_widget *w,
 		soc_widget_read(w, e->reg, &val);
 		item = (val >> e->shift_l) & e->mask;
 
-		if (item < e->items && !strcmp(p->name, e->texts[item]))
+		if (item < e->items &&
+			!strcmp(p->name, snd_soc_get_enum_text(e, item)))
 			p->connect = 1;
 		else
 			p->connect = 0;
@@ -584,7 +585,7 @@ static void dapm_set_path_status(struct snd_soc_dapm_widget *w,
 		 * that the default mux choice (the first) will be
 		 * correctly powered up during initialization.
 		 */
-		if (!strcmp(p->name, e->texts[0]))
+		if (!strcmp(p->name, snd_soc_get_enum_text(e, 0)))
 			p->connect = 1;
 	}
 	break;
@@ -600,7 +601,8 @@ static void dapm_set_path_status(struct snd_soc_dapm_widget *w,
 				break;
 		}
 
-		if (item < e->items && !strcmp(p->name, e->texts[item]))
+		if (item < e->items &&
+			!strcmp(p->name, snd_soc_get_enum_text(e, item)))
 			p->connect = 1;
 		else
 			p->connect = 0;
-- 
1.9.1

